client/src/area.rs

```diff
@@ -51,3 +55,48 @@ impl Into<Area> for Size {
         }
     }
 }
+
+#[derive(Debug, Default, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Margin {
```

Comment:
layer-surface calls it margin, awesome calls it strut. I think margin is more natural.
If someone knows why the layer-surface protocol allows for negative values ( https://github.com/swaywm/wlr-protocols/blob/bc1364930c5f7a309b791403c05d0ca7489b4b07/unstable/wlr-layer-shell-unstable-v1.xml#L171-L174 ) this should be added as a comment here!

Reply:
According to @emersion on #sway-devel "the original idea was to allow slide-out animations"